src/commands/crashes/upload-symbols.ts

```diff
@@ -0,0 +1,420 @@
+      }
+
+      // Checking if source map file points to the existing file
+      let sourceMapFileStats: Fs.Stats
+      try {
+        sourceMapFileStats = Fs.statSync(path)
+      } catch (error) {
+        throw new UploadSymbolsError(ErrorCodes.InvalidParameter, `Can't open file ${path}: ${this.getErrnoExceptionString(error)}`);
+      }
+
+      // Checking if source map file is actually a file
+      if(!sourceMapFileStats.isFile()){
+        throw new UploadSymbolsError(ErrorCodes.InvalidParameter, `${path} is not a file`);
+      }
+
+      // Adding (or replacing) source map file
+      const sourceMapFileBaseName = Path.basename(path);
+      let existingFile = zip.getEntry(sourceMapFileBaseName);
+      if(!_.isNil(existingFile)){    
+        debug("Removing existing mappings file with the same name");
+        zip.deleteFile(sourceMapFileBaseName);
+      }
+      debug("Adding the specified mappings file to the zip");
+      try {
+        zip.addLocalFile(path);
+      } catch (error) {
+        throw new UploadSymbolsError(ErrorCodes.Exception, `unable to add file ${path} to the zip`);
+      }
+  }
+
+  private getZipBuffer(zip: AdmZip): Buffer{
+    try {
+      debug("Getting in-memory ZIP archive as Buffer");
+      return zip.toBuffer();
+    } catch (error){
+       throw new UploadSymbolsError(ErrorCodes.Exception, `Failed to compress the ZIP file: ${_.toString(error)}`);
+    }
+  }
+
+  private prepareZipFromXcArchive(path: string): AdmZip {
+    debug(`Trying to prepare the ZIP archive with symbols from .xcarchive folder`);
+    const fsEntryStats = this.getStatsForFsPath(path);
+    const symbolsType = this.detectSymbolsFsEntryType(path, fsEntryStats);
+    switch(symbolsType){
+      case SymbolFsEntryType.XcArchive:
+        // The DSYM folders from "*.xcarchive/dSYMs" should be compressed
+        const dsymsFolderPath = Path.join(path, "dSYMs");
+        return this.packDsymParentFolderContents(dsymsFolderPath);
+      default:
+        // File doesn't points to correct .xcarchive
+        throw new UploadSymbolsError(ErrorCodes.InvalidParameter, `${path} is not a valid XcArchive folder`);
+    }
+  }
+
+  private async abortUploadingRequest(client: MobileCenterClient, app: DefaultApp, symbolUploadId: string): Promise<models.SymbolUpload> {
+    debug("Uploading failed, aborting upload request");
+    try {
+      return await this.executeSymbolsUploadingEndRequest(client, app, symbolUploadId, "aborted");
+    }
+    catch(ex){
+      debug('Failed to correctly abort the uploading request');
+      out.text(`Symbols Uploading Abort failed`);
+    }
+  }
+}
+
+type SymbolsUploadEndRequestStatus = 'committed' | 'aborted';
+
+enum SymbolFsEntryType {
+  Unknown,
+  DsymFolder,
+  DsymParentFolder,
+  XcArchive,
+  ZipFile
+}
+
+class UploadSymbolsError extends Error {
+  constructor(public errorCode: ErrorCodes, message: string){
```

Comment:
Yes, it is stored, I have used some syntactic sugar here - e.g. https://jsfiddle.net/gawg9ru5/
It is called "Parameter properties" in [TS handbook](https://www.typescriptlang.org/docs/handbook/classes.html)